tests/shakedown/shakedown/clients/node.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+import logging
+import os
+import requests
+import urllib
+
+from . import dcos_url_path
+from .authentication import dcos_acs_token, DCOSAcsAuth
+from .rpcclient import verify_ssl
+
+logger = logging.getLogger(__name__)
+
+
+class Collection(ABC):
+    """Resources have collections of objects."""
+
+    @abstractmethod
+    def create(self):
+        pass
+
+
+class Session(ABC):
+
+    def create_url(self, path):
+        """Create the URL based off this partial path."""
+        return urllib.parse.urljoin(self.base_url, path)
+
+    def get(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        kwargs['verify'] = verify_ssl()
+        return requests.get(url, *args, **kwargs)
+
+    def post(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        kwargs['verify'] = verify_ssl()
+        return requests.post(url, *args, **kwargs)
+
+
+class DiagnosticBundle():
+
+    def __init__(self, session, bundle_name):
+        logger.info('Created diagnostic bundle %s', bundle_name)
+
+        self.session = session
+        self.bundle_name = bundle_name
+
+    def status(self):
```

Comment:
could we rename this? `status` returning filename or none was a bit confusing to me

Reply:
Sure. The diagnostics API is a little weird and I was unsure how the client should look like. What do you think about `path` or `download_path` returning the download URL or `None`?